pkg/packagekit/applenotarization/applenotarization.go

```diff
@@ -52,29 +62,61 @@ func (n *Notarizer) Submit(ctx context.Context, filePath string, primaryBundleId
 	// TODO check file extension here
 	// zip,pkg,dmg
 
-	response, err := n.runAltool(ctx, []string{
-		"--notarize-app",
-		"--primary-bundle-id", primaryBundleId,
-		"--file", filePath,
-	})
+	// Sometimes submit fails. Retry a couple times.
+	for attempt := 1; attempt <= maxSubmitTries; attempt++ {
+
+		// According to a support thread on apple, the bundle-id passed to
+		// altool's notarize is unrelated to the actual software. It is only
+		// used to for something in local storage. As such, add a nonce to
```

Comment:
this is interesting - is the `--primary-bundle-id` a required flag despite this?

Reply:
Yep. required flag. But, as the comment says, eskimo1 says it's just for local tool caching.